tsdb/engine/wal/wal.go

```diff
@@ -455,7 +458,7 @@ func (l *Log) nextMetaFile() error {
 			return err
 		}
 
-		id = int(n)
+		id = int(n) + 1
```

Comment:
So the locking *and* this change are key to " Fix metafile so it doesn't get trampled by other goroutines"? Or are there two issues here, but only 1 in the title?